_posts/2019-07-18-edge-to-kubernetes.md

```diff
@@ -0,0 +1,125 @@
+
+## Configuring mTLS for the Ingress Controller
+
+Because the Nginx Ingress controller is already configured to support mTLS and subsequently does all the heavy-lifting, all we really need to do is to tell our Ingress object to use the built-in functionalities.
+
+Before we do this however, we first need to download the WoTT CA and upload this to the cluster. Otherwise Nginx will not be able to verify the certificate.
+
+To do this, let's assume that our application will run in the 'foobar' namespace within the cluster.
+
+
+Create the namespace if it doesn't already exist:
+
+```
+$ kubectl create namespace foobar
+```
+
+Download the WoTT CA certificate bundle:
+```
+$ curl -s https://api.wott.io/v0.2/ca-bundle | \
+  jq -r .ca_bundle > wott-ca.crt
+```
+
+Upload the CA certificate to the cluster:
+
+```
+$ kubectl create secret generic wott-ca \
+  -n foobar \
+  --from-file=ca.crt=wott-ca.crt
+```
+
+We can verify that we the secret was properly created by running:
+
+```
+$ kubectl get secrets -n foobar
+NAME                  TYPE                                  DATA      AGE
+[...]
+wott-ca               Opaque                                1         4s
+```
+
+With CA in place, we can move on to configuring the [Ingress object](https://kubernetes.io/docs/concepts/services-networking/ingress/). The Ingress object is responsible for informing the Ingress Controller that it should route traffic addressed to a given hostname to a particular service.
+
+Here's an example of the YAML file for such configuration (here we are using cert-manager for the SSL configuration):
+
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+    certmanager.k8s.io/acme-challenge-type: http01
+    certmanager.k8s.io/issuer: letsencrypt-prod
+    kubernetes.io/ingress.class: nginx
+    nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: "true"
+    nginx.ingress.kubernetes.io/auth-tls-secret: foobar/wott-ca
+    nginx.ingress.kubernetes.io/auth-tls-verify-client: "on"
+    nginx.ingress.kubernetes.io/auth-tls-verify-depth: "2"
+  name: mtls-test
+  namespace: api
+spec:
+  rules:
+  - host: mtls-test.example.com
+    http:
+      paths:
+      - backend:
+          serviceName: foobar
+          servicePort: 80
+        path: /?(.*)
+  tls:
+  - hosts:
+    - mtls-test.example.com
+    secretName: le-secret-mtls-test
+```
+
+The secret for mTLS lies in the varies 'auth-tls' annotations that we have above. These annotations tell Nginx to configure auth-tls (i.e. mTLS) for this endpoint.
+
+We can now leverage the HTTP headers that Nginx is passing on to the appserver(s) just like we did in our [last blog post]({{site.url}}/blog/tutorials/2019/07/15/mtls-with-nginx).
```

Comment:
maybe specify that you mean the mtls with nginx example rather than just 'last post'